### DISCUSSION TITLE
RuntimeError: "spmm" not implemented for 'Half'

### DISCUSSION THREAD

**Discussion by kfoynt** [snap-stanford/ogb#205] [MAG240M-LSC]
I am getting this error from the baseline code that seems to be related to cuSPARSE not supporting half-precision https://docs.nvidia.com/cuda/cusparse/index.html
Which libraries have you used to run the baseline?

**Comment by rusty1s** [comment]
You need to update to the latest `torch-sparse` release.

**Reply by kfoynt** [reply]
Thanks. I did not reply earlier because I was trying to fix that. I am using torch 1.7.1 and cuda 11, I upgraded torch-sparse to 0.6.9.
However, I am now getting the following error:
`---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-7a0d60825345> in <module>
     21                                seed_everything)
     22 
---> 23 from torch_sparse import SparseTensor
     24 from torch_geometric.nn import SAGEConv, GATConv
     25 from torch_geometric.data import NeighborSampler

~/berserker/lib/python3.7/site-packages/torch_sparse/__init__.py in <module>
     13 ]:
     14     torch.ops.load_library(importlib.machinery.PathFinder().find_spec(
---> 15         f'{library}_{suffix}', [osp.dirname(__file__)]).origin)
     16 
     17 if torch.cuda.is_available():  # pragma: no cover

AttributeError: 'NoneType' object has no attribute 'origin'`

I looked online a bit and found 
https://github.com/rusty1s/pytorch_sparse/issues/127 
and 
https://github.com/rusty1s/pytorch_geometric/issues/2304 

I seem to have the same issue. the cuda files are not created in the torch_sparse folder. 
I tried using python 3.7 and 3.8, I updated all other torch_* to their latest versions, I checked if there are multiple versions of each. Unfortunately, nothing worked.

**Reply by kfoynt** [reply]
Alright. I just got it working. Probably just pure luck. 

I think a combination of python 3.8 and
`pip install torch-sparse -f https://pytorch-geometric.com/whl/torch-1.7.0+cu102.html
`
must have made it work. 

Previously, I was trying python 3.8 or 3.7 and 
`pip install torch-sparse==0.6.9 -f https://pytorch-geometric.com/whl/torch-1.7.1+cu102.html`
